test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_smoke.py

```diff
@@ -69,7 +69,13 @@ def setup_method(self, method):
                         "best_gap": 0.000000, "last_t": 5,
                         "best_t": 5, "disp": 0.005000,
                         "error": 0.442883, "n_oracle_calls": 19,
-                        "n_predictors": 6}]
+                        "n_predictors": 6},
+                       {"cons_class": TruePositiveRateDifference, "eps": 0.005,
+                        "best_gap": 0.0, "last_t": 5,
```

Comment:
This test example looks too easy to solve for the algorithm (the fairness constraints are satisfied automatically). Please try to run it on a version of the data where the label 'y' is flipped... in that case you should obtain the error that matches the one obtained for `EqualizedOdds` with the same `eps`.

Reply:
So does this mean that for `y=1`, there is already no disparity in the test data? So what we should check for is for `y=0`?

Reply:
@MiroDudik  I ran a basic test as in [this gist](https://gist.github.com/hannanabdul55/536a1a9f25a6c39fd60f860bbd0938b5#file-test_exponentiatedgradient_smoke-py-L118). But I did not get the same error value. This is what I got after running the test: 
```python
1. Error: 0.25 Disparity: 0.0050000000000000044
Flipped y
2. Error: 0.4271333333333335 Disparity: 0.004999999999999893
```
- The first set of values is for the original generated data. 
- The second set is for the data with the `y`'s flipped. `list(map(lambda i: int(not i), labels))`. 
I did not understand how the error could be better for either case.  
PS: The error for the Equalized Odds case is `0.442883`(as per the smoke test result for eps=`0.05`.)

Reply:
This might still be okay. Can you try to run `ExponentiatedGradient` on the flipped labels as well?

Also, try to evaluate both the **equalized odds** based disparity as well as the TPR Difference based disparity in both cases (in other words, you can just cycle through the two `disparity_moment` settings).

Reply:
These are the values I get for the flipped case,
```python
{"cons_class": TruePositiveRateDifference, "eps": 0.005,
         "best_gap": 0.0, "last_t": 5,
         "best_t": 5, "disp": 0.005000,
         "error": 0.427133, "n_oracle_calls": 17,
         "n_predictors": 3},
{"cons_class": EqualizedOdds, "eps": 0.005,
         "best_gap": 0.000000, "last_t": 5,
         "best_t": 5, "disp": 0.005000,
         "error": 0.442883, "n_oracle_calls": 19,
         "n_predictors": 6}
```
I think from this, it makes sense because for the same value of `eps`, the error introduced would be more for `EqualizedOdds` than TPR as it tried to account for the fairness in both the values of `y`,rather than just `y=1`. In this process, it injects more error? Could this be a reasonable explanation for this?